playground/didchat2/src/lib.rs

```diff
@@ -0,0 +1,298 @@
+    match config.cmd {
+        CMD::Init{ path } => init(&path),
+        CMD::Doc{ path } => doc(&path),
+        CMD::Did{ path } => did(&path),
+        CMD::Connect{ path, name, did } => connect(&path, &name, &did),
+        CMD::Send{ path, name, message } => send(&path, &name, &message),
+        CMD::Receive{ path, encrypted_message } => receive(&path, &encrypted_message),
+        CMD::Help => help()
+    }
+}
+
+/**
+ * new - Creates a public/private key-pair if does not already exists, effectively creating a new chat.
+ */ 
+fn init(path: &str) -> Result<String, std::io::Error> {
+    use std::io::Write;
+
+    // 1. Create empty folders
+    if !std::fs::metadata(root_path(path)).is_ok() {
+        std::fs::create_dir_all(path).unwrap_or_default();
+    }
+
+    if !std::fs::metadata(names_path(path)).is_ok() {
+        std::fs::create_dir_all(names_path(path)).unwrap_or_default();
+    }
+
+    if !std::fs::metadata(dids_path(path)).is_ok() {
+        std::fs::create_dir_all(dids_path(path)).unwrap_or_default();
+    }
+
+    if !std::fs::metadata(seed_path(path)).is_ok() {
+        // 2. Generate seed
+        let mut csprng = rand_core::OsRng{};
+        let seed = ed25519_dalek::SecretKey::generate(&mut csprng);
+        let seed_bytes = seed.as_bytes();
+
+        // 3. Write seed to file
+        let mut file = std::fs::File::create(seed_path(path)).unwrap();
+        file.write(seed_bytes).unwrap();
+        
+        Ok(format!("{} is ready", path))
+    } else {
+        Ok(format!("{} already exists", path))
+    }
+}
+
+fn doc(path: &str) -> Result<String, std::io::Error> {
+    use did_key::DIDCore;
+    use did_key::KeyMaterial;
+    
+    // 1. Read seed from file
+    let seed = std::fs::read(seed_path(path)).unwrap();
+
+    // 2. Transform seed to a did-document
+    let public_private_keypair = did_key::Ed25519KeyPair::from_seed(&seed);
+    let public_only_keypair = did_key::from_existing_key::<did_key::Ed25519KeyPair>(&public_private_keypair.public_key_bytes(), None);
+    let doc = public_only_keypair.get_did_document(did_key::CONFIG_LD_PUBLIC);
+
+    // 3. Serialize did-document to json
+    let did_document = serde_json::to_string_pretty(&doc).unwrap();
+    Ok(format!("{}", did_document))
+}
+
+fn did(path: &str) -> Result<String, std::io::Error> {
+    use did_key::DIDCore;
+
+    // 1. Read seed from file
+    let seed = std::fs::read(seed_path(path)).unwrap();
+
+    // 2. Transform seed to a did
+    let keypair = did_key::Ed25519KeyPair::from_seed(&seed);
+    let diddoc: did_key::Document = keypair.get_did_document(did_key::CONFIG_LD_PUBLIC);
+    let did = diddoc.id;
+
+    // 3. Print did
+    Ok(format!("{}", did))
+}
+
+fn connect(path: &str, name: &str, did: &str) -> Result<String, std::io::Error> {
+    use std::io::Write;
+
+    // 2. Create 'name' -> 'did' mapping
+    let mut file = std::fs::File::create(name_path(path, name)).unwrap();
+    file.write(did.as_bytes()).unwrap();
```

Comment:
replace `unwrap` with a question mark `?`

Do it for all the IO operations that can throw an error.